### DISCUSSION TITLE
Project Board lags/hangs after every edit

### DISCUSSION THREAD

**Discussion by allytsj** [community/community#12483] [Projects and Issues]
On the project board, every time I try to make an edit to any card (e.g. add tag, close issue, move card),
the page will lag or hang (i.e. I'll get the "page is unresponsive" error message on chrome), and I either need to wait for the page to become responsive or refresh the page. 

I'm an outside collaborator of this board, it used to be working fine when my access was restricted to just opening and closing issues, but since I've been given more access to moving the cards it started to become super laggy and unresponsive. 

Is there any way I can resolve this?

**Comment by bartgrundeken** [comment]
It lags for me to the point where it is unworkable. However, I only have this issue on Chrome/Ubuntu 20. On Firefox/Ubuntu 20 it works fine.

**Reply by bartgrundeken** [reply]
Update: issue also happens on MacOs Ventura with Chrome. Notable even with a M1 Max with 64 gb internal memory.

**Comment by drdbuck** [comment]
I also have the issue! I use chrome.

**Reply by drdbuck** [reply]
How can we report this problem?

**Reply by bartgrundeken** [reply]
Keep bumping this issue, I guess

**Reply by andreasfunnel** [reply]
Me, and other colleagues, also have this issue. It's unbearable at this point

**Comment by LoneDev6** [comment]
This is one of the few reasons why I can't use projects.
Instead of making my workflow easier and faster it just slows down my whole daily work.
It's easier to manage stuff using milestones and tags at this point.

**Comment by lsagetlethias** [comment]
As far as I tested, it looks like using iteration based filters with `@next / @current / @previous` is causing my project tabs to lag.

Removing it fix the hanging on edit, open ticket, changing status, etc.

Hope this help.

**Comment by LoneDev6** [comment]
The whole projects logic must be optimized.
Right now it's not usable for real life projects, only for hobby projects with 10-15 issues.

Test made on a PC with these specs (can easily play latest videogames and software without lag).
CPU: i7-4790k
GPU: GTX 980 TI
RAM: 32 GB

Tested with different browsers, without extensions and with clean profiles, same issue.

https://user-images.githubusercontent.com/27242001/232096324-52b5af2b-04a2-480c-893a-579f6b8ac09a.mp4

**Comment by kashanghori** [comment]
Quick note: it's working fine on firefox, the bug seems to be on chrome where it lags taking 100% resources

**Comment by john-thomas-dotcms** [comment]
I also have this issue.

Where/when:
* On Brave (Chromium-based) on Windows 10.
* It used to work without problems, but started lagging on me a couple months ago.
* The problem persisted through a complete OS and app reinstall.
* The problem happens when no browser extensions are enabled.
* It only happens when opening an issue from the project board, and then editing a comment in the issue.
    * If I open the issue in its own window, I have no issues.
    * If I edit other fields in the issue popup, it doesn't freeze.
    * I'm not sure if it will freeze if I add a new comment (instead of editing the original one) because I very rarely do that in my workflow.

Symptoms/Severity:
* The lag is really a freeze that happens after typing the first characters in the comment; I get 6-20 characters typed normally, and then it stops dead (typing stopped, and no mouse action available anywhere else in the issue popup).
* The lag is extreme - sometimes it freezes for a half an hour before it becomes usable again.
* After it comes back from the initial lag, it allows editing normally for a while, but then randomly freezes again.
    * The later freezes usually last only 10-60 seconds, instead of up to half an hour.
